frontend/ui/src/pages/Channels/ConnectedChannelsList/ChannelListItem.tsx

```diff
@@ -0,0 +1,151 @@
+import {ReactComponent as WhatsappLogo} from 'assets/images/icons/whatsapp_avatar.svg';
+import {ReactComponent as AiryLogo} from 'assets/images/icons/airy_avatar.svg';
+import {ReactComponent as CheckMark} from 'assets/images/icons/checkmark.svg';
+import styles from './ChannelListItem.module.scss';
+import {RouteComponentProps, withRouter} from 'react-router-dom';
+import {CHANNELS_CHAT_PLUGIN_ROUTE, CHANNELS_FACEBOOK_ROUTE} from '../../../routes/routes';
+
+type ChannelItemProps = {
+  channel: Channel;
+} & ConnectedProps<typeof connector> &
+  RouteComponentProps<{channelId: string}>;
+
+const mapDispatchToProps = {
+  disconnectChannel,
+};
+
+const connector = connect(null, mapDispatchToProps);
+
+const ChannelItem = (props: ChannelItemProps) => {
+  const {channel} = props;
+  const [deletePopupVisible, setDeletePopupVisible] = useState(false);
+
+  const disconnectChannelRequestPayload = {
+    channelId: channel.id,
+  };
+
+  const channelIcon = (channel: Channel) => {
+    switch (channel.source) {
+      case ChannelSource.facebook:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <FacebookLogo />;
+        }
+      case ChannelSource.google:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <GoogleLogo />;
+        }
+      case ChannelSource.twilioSMS:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <SMSLogo />;
+        }
+      case ChannelSource.twilioWhatsapp:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <WhatsappLogo />;
+        }
+      case ChannelSource.chatPlugin:
+        if (channel?.metadata?.imageUrl) {
+          return <img className={styles.imageUrlLogo} src={channel.metadata.imageUrl} />;
+        } else {
+          return <AiryLogo />;
+        }
+      default:
+        return <AiryLogo />;
+    }
+  };
+
+  const editChannel = () => {
+    switch (channel.source) {
+      case ChannelSource.facebook:
+        return {pathname: CHANNELS_FACEBOOK_ROUTE + `/${channel.id}`, state: {channel: channel}};
+      case ChannelSource.google:
+        break;
+      case ChannelSource.twilioSMS:
+        break;
+      case ChannelSource.twilioWhatsapp:
+        break;
+      case ChannelSource.chatPlugin:
+        return {pathname: CHANNELS_CHAT_PLUGIN_ROUTE + `/${channel.id}`, state: {channel: channel}};
+    }
+  };
+
+  const disconnectChannel = (channelSource: string) => {
+    let source: string;
+    if (channelSource === 'chat_plugin') {
+      source = 'chatplugin';
+    } else {
+      source = channel.source;
+    }
```

Comment:
a sign we should probably do something about the naming here to avoid those conditionals (we control the names after all)

Reply:
Agree 100% this doesn't look pretty